### PR TITLE
Convert SCSI controller functional test to unit test

### PR DESCRIPTION
**Pull request by Barakmor1**
This functional test is expensive and mainly verify libvirt behavior, which is out of our scope. Our responsibility is to ensure correct conversion of the VMI to the domain spec.

Replace the functional test with a unit test that directly verifies the domain conversion logic with and without hotplug enabled.

/kind cleanup

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least one e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by Barakmor1** [comment]
This functional test showed a [strange flake](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/15224/pull-kubevirt-e2e-k8s-1.32-sig-compute/1946231180277846016), but it does not seem related to the test logic itself. It looks more like a node issue, possibly a bug or overload, that caused the VMI to get stuck in the scheduling phase for some time. If you have any insights about this flake or know whether it has already been fixed, please share

FYI @jean-edouard  @lyarwood @awels

File: pkg/virt-launcher/virtwrap/converter/converter_test.go [hunk 1]
```diff
@@ -803,6 +803,37 @@ var _ = Describe("Converter", func() {
...
+						Expect(controller.Model).To(Equal(model))
+					}
+				}
+				Expect(found).To(Equal(enabled))
+			},
+				Entry("should appear if enabled for amd64", true, amd64),
+				Entry("should appear if enabled for ppc64le", true, ppc64le),
+				Entry("should appear if enabled for arm64", true, arm64),
+				Entry("should appear if enabled for s390x", true, s390x),
+				Entry("should NOT appear if disabled for amd64", false, amd64),
```
Comment (0xFelix) [inline pkg/virt-launcher/virtwrap/converter/converter_test.go]: What about other arches when disabled?

**Review by 0xFelix — COMMENTED** [review]
/approve

File: pkg/virt-launcher/virtwrap/converter/converter_test.go (hunk 1 above)
Comment (Barakmor1) [reply pkg/virt-launcher/virtwrap/converter/converter_test.go]: I was thinking about it, and honestly, I’m not sure if we should add these combinations since they seem less interesting. I was also a bit worried it might be too much boilerplate. 
let me know if you think we should include them. I’m happy to add them.

Comment (0xFelix) [reply pkg/virt-launcher/virtwrap/converter/converter_test.go]: If we make an effort to add the positive test case for all arches, why not do it for the negative test case as well. Only adds a few more lines.

Comment (Barakmor1) [reply pkg/virt-launcher/virtwrap/converter/converter_test.go]: Done

File: pkg/virt-launcher/virtwrap/converter/converter_test.go [hunk 2]
```diff
@@ -803,6 +803,40 @@ var _ = Describe("Converter", func() {
 			//TODO add s390x entry with custom check of model used (disks/interfaces/controllers/devices will use different models)
 		)
 
+		Context("with ephemeral disk", func() {
+			DescribeTable("a scsi controller", func(enabled bool, arch string) {
```
Comment (lyarwood) [inline pkg/virt-launcher/virtwrap/converter/converter_test.go]: nit - Passing a `GomegaMatcher` here instead of the `bool` would be cleaner but not required.

```go
DescribeTable("a scsi controller", func(match types.GomegaMatcher, arch string) {
[..]
  Expect(domain.Spec.Devices.Controllers).To(match)
[..]
},
  Entry("should appear if enabled for amd64",
    ContainElement(
      api.Controller{
        Type: "scsi",
        Index: "0",
        Model: "virtio-non-transitional",
      },
    ),
    amd64,
  ),
)
```

File: pkg/virt-launcher/virtwrap/converter/converter_test.go [hunk 3]
```diff
@@ -803,6 +803,40 @@ var _ = Describe("Converter", func() {
...
+				for _, controller := range domain.Spec.Devices.Controllers {
+					if controller.Type == "scsi" {
+						found = true
+						Expect(controller.Index).To(Equal("0"))
+						Expect(controller.Model).To(Equal(model))
+					}
+				}
+				Expect(found).To(Equal(enabled))
+			},
+				Entry("should appear if enabled for amd64", true, amd64),
```
Comment (lyarwood) [inline pkg/virt-launcher/virtwrap/converter/converter_test.go]: supernit - move the `should` into the `DescribeTable` description.

**Comment by lyarwood** [comment]
> This functional test showed a [strange flake](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/15224/pull-kubevirt-e2e-k8s-1.32-sig-compute/1946231180277846016), but it does not seem related to the test logic itself. It looks more like a node issue, possibly a bug or overload, that caused the VMI to get stuck in the scheduling phase for some time. If you have any insights about this flake or know whether it has already been fixed, please share

Yeah no idea why the flake appeared in the first place, looking at the logs the `launcher` pod does schedule eventually but the `volumedisk0` container isn't ready when we capture the logs. That does suggest we've overloaded the env somehow causing a delay in scheduling but I have to admit this is the first time I've seen this recently. Something to keep an eye out for over the next few weeks: 

```
$ curl https://storage.googleapis.com/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/15224/pull-kubevirt-e2e-k8s-1.32-sig-compute/1946231180277846016/artifacts/k8s-reporter/4/1_pods.log | jq ' .items.[] | select (.metadata.name | contains("testvmi-zwjmk")) | .metadata.name,.status.conditions '
[..]
"virt-launcher-testvmi-zwjmk-65l8f"
[
  {
    "type": "kubevirt.io/virtual-machine-unpaused",
    "status": "True",
    "lastProbeTime": "2025-07-18T16:05:26Z",
    "lastTransitionTime": "2025-07-18T16:05:26Z",
    "reason": "NotPaused",
    "message": "the virtual machine is not paused"
  },
  {
    "type": "PodReadyToStartContainers",
    "status": "True",
    "lastProbeTime": null,
    "lastTransitionTime": "2025-07-18T16:05:27Z"
  },
  {
    "type": "Initialized",
    "status": "True",
    "lastProbeTime": null,
    "lastTransitionTime": "2025-07-18T16:05:30Z"
  },
  {
    "type": "Ready",
    "status": "False",
    "lastProbeTime": null,
    "lastTransitionTime": "2025-07-18T16:05:26Z",
    "reason": "ContainersNotReady",
    "message": "containers with unready status: [compute volumedisk0]"
  },
  {
    "type": "ContainersReady",
    "status": "False",
    "lastProbeTime": null,
    "lastTransitionTime": "2025-07-18T16:05:26Z",
    "reason": "ContainersNotReady",
    "message": "containers with unready status: [compute volumedisk0]"
  },
  {
    "type": "PodScheduled",
    "status": "True",
    "lastProbeTime": null,
    "lastTransitionTime": "2025-07-18T16:05:26Z"
  }
]
```

File: pkg/virt-launcher/virtwrap/converter/converter_test.go (hunk 2 above)
Comment (Barakmor1) [reply pkg/virt-launcher/virtwrap/converter/converter_test.go]: Done some similar refactoring :)

File: pkg/virt-launcher/virtwrap/converter/converter_test.go (hunk 3 above)
Comment (Barakmor1) [reply pkg/virt-launcher/virtwrap/converter/converter_test.go]: Done
